src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -0,0 +1,86 @@
+using System.Runtime.InteropServices;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class OasdiffManager : IOasdiffManager
+{
+    private const string OasdiffVersion = "1.9.2";
+
+    private readonly IHttpClientWrapper _httpClientWrapper;
+    private readonly IProcessWrapper _processWrapper;
+    private readonly string _toolDirectory;
+
+    public OasdiffManager(IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    {
+        this._httpClientWrapper = httpClientWrapper;
+        this._processWrapper = processWrapper;
+        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+    }
+
+    public async Task InstallOasdiffAsync(CancellationToken cancellationToken)
+    {
+        Directory.CreateDirectory(this._toolDirectory);
+
+        var oasdiffFileName = GetOasdiffFileName();
+        var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
+
+        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._toolDirectory, oasdiffFileName), cancellationToken);
+        await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
+    }
+
+    private async Task DecompressDownloadedFileAsync(string oasdiffFileName, CancellationToken cancellationToken)
+    {
+        var pathToCompressedFile = Path.Combine(this._toolDirectory, oasdiffFileName);
+        await this._processWrapper.RunProcessAsync("tar", new[] { "-xzf", $"{pathToCompressedFile}", "-C", $"{this._toolDirectory}" }, cancellationToken);
+    }
+
+    private static string GetOasdiffFileName()
+    {
+        var osType = GetOperatingSystem();
+        var architecture = GetArchitecture();
+
+        var fileName = $"oasdiff_{OasdiffVersion}_{osType}_{architecture}.tar.gz";
+
+        if (osType == "macos")
+        {
+            fileName = $"oasdiff_{OasdiffVersion}_darwin_all.tar.gz";
+        }
+
+        return fileName;
+    }
+
+    private static string GetOperatingSystem()
```

Comment:
I think GetOperatingSystem and GetArchitecture are both used in OasdiffManager and SpectralManager, they could probably be in some common class?

Reply:
No they have different implementation since their use is to build the name of the file. Both Oasidiff and spectral use a different naming pattern. For instance, instead of using x64 they use amd64 (synomym).